packages/flutter/lib/src/services/raw_keyboard_android.dart

```diff
@@ -0,0 +1,205 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'raw_keyboard.dart';
+
+/// Platform-specific key event data for Android.
+///
+/// This object contains information about key events obtained from Android's
+/// `KeyEvent` interface.
+///
+/// See also:
+///
+///  * [RawKeyboard], which uses this interface to expose key data.
+class RawKeyEventDataAndroid extends RawKeyEventData {
+  /// Creates a key event data structure specific for Android.
+  ///
+  /// The [flags], [codePoint], [keyCode], [scanCode], and [metaState] arguments
+  /// must not be null.
+  const RawKeyEventDataAndroid({
+    this.flags = 0,
+    this.codePoint = 0,
+    this.keyCode = 0,
+    this.scanCode = 0,
+    this.metaState = 0,
+  })  : assert(flags != null),
+        assert(codePoint != null),
+        assert(keyCode != null),
+        assert(scanCode != null),
+        assert(metaState != null);
+
+  /// The current set of additional flags for this event.
+  ///
+  /// Flags indicate things like repeat state, etc.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getFlags()>
+  /// for more information.
+  final int flags;
+
+  /// The Unicode code point represented by the key event, if any.
+  ///
+  /// If there is no Unicode code point, this value is zero.
+  ///
+  /// Dead keys are represented as Unicode combining characters.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getUnicodeChar()>
+  /// for more information.
+  final int codePoint;
+
+  /// The hardware key code corresponding to this key event.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getKeyCode()>
+  /// for more information.
+  final int keyCode;
+
+  /// The hardware scan code corresponding to this key event.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getScanCode()>
+  /// for more information.
+  final int scanCode;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See <https://developer.android.com/reference/android/view/KeyEvent.html#getMetaState()>
+  /// for the numerical values of the `metaState`. Many of these constants are
+  /// also replicated as static constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * The [modifiersPressed] accessor to get a Set of currently pressed
```

Comment:
nit: the style for "see also" always puts the linked property first followed by a short explanation.